libjlm/include/jlm/ir/operators/alloca.hpp

```diff
@@ -61,38 +74,33 @@ class alloca_op final : public jive::simple_op {
 
 	static std::unique_ptr<jlm::tac>
 	create(
-		const jive::type & vtype,
+		const jive::valuetype & allocatedType,
 		const variable * size,
 		size_t alignment)
 	{
-		auto vt = dynamic_cast<const jive::valuetype*>(&vtype);
-		if (!vt) throw jlm::error("expected value type.");
-
 		auto bt = dynamic_cast<const jive::bittype*>(&size->type());
 		if (!bt) throw jlm::error("expected bits type.");
 
-		alloca_op op(PointerType(*vt), *bt, alignment);
+		alloca_op op(allocatedType, *bt, alignment);
 		return tac::create(op, {size});
 	}
 
 	static std::vector<jive::output*>
 	create(
-		const jive::type & type,
+		const jive::valuetype & allocatedType,
 		jive::output * size,
 		size_t alignment)
 	{
-		auto vt = dynamic_cast<const jive::valuetype*>(&type);
-		if (!vt) throw jlm::error("expected value type.");
-
 		auto bt = dynamic_cast<const jive::bittype*>(&size->type());
 		if (!bt) throw jlm::error("expected bits type.");
 
-		alloca_op op(PointerType(*vt), *bt, alignment);
+		alloca_op op(allocatedType, *bt, alignment);
 		return jive::simple_node::create_normalized(size->region(), op, {size});
 	}
 
 private:
 	size_t alignment_;
+  std::unique_ptr<jive::type> AllocatedType_;
```

Comment:
Why isn't this a jive::valuetype?

Reply:
The copy method used in the constructor is returning std::unique_ptr<jive::type>, and there it is not very readable to change the type to a jive::valuetype there. So I decided to rather change it in the value_type() method.